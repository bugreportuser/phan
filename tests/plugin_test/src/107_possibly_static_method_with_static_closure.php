<?php
// @phan-file-suppress PhanPluginUndeclaredVariableIsset

class PSMClazz107 {
    public function examplePublic() {
        (static function() {
            var_export(isset($this));
        })();
    }

    protected function exampleProtected() {
        (static function() {
            var_export(isset($this));
        })();
    }

    private function examplePrivate() {
        (static function() {
            var_export(isset($this));
        })();
    }

    public function main() {
        $this->examplePublic();
        $this->exampleProtected();
        $this->examplePrivate();
    }
}

(new PSMClazz107())->main();
