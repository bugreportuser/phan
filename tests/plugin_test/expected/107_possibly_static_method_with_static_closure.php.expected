src/107_possibly_static_method_with_static_closure.php:5 PhanPluginPossiblyStaticPublicMethod Public method \PSMClazz107::examplePublic can be static
src/107_possibly_static_method_with_static_closure.php:11 PhanPluginPossiblyStaticProtectedMethod Protected method \PSMClazz107::exampleProtected can be static
src/107_possibly_static_method_with_static_closure.php:17 PhanPluginPossiblyStaticPrivateMethod Private method \PSMClazz107::examplePrivate can be static
